contrib/gorm.io/gorm.v1/gorm.go

```diff
@@ -121,5 +121,8 @@ func after(db *gorm.DB, operationName string, cfg *config) {
 	}
 
 	span, _ := tracer.StartSpanFromContext(ctx, operationName, opts...)
-	span.Finish(tracer.WithError(db.Error))
+	defer span.Finish()
+	if cfg.errCheck(db.Error) {
+		span.SetTag(ext.Error, db.Error)
```

Comment:
Is there a reason for using `SetTag()` instead of `tracer.WithError()` here? If not, the latter would probably be preferable as it reduces the burden on the reviewer to verify that the two are interchangable.

Reply:
@felixge Maybe there ref https://github.com/DataDog/dd-trace-go/blob/63dea358a04d495f0b0f25b363b77eabe4e09153/contrib/jinzhu/gorm/gorm.go#L141

Reply:
I changed to:

```go
	var dbErr error
	if cfg.errCheck(db.Error) {
		dbErr = db.Error
	}
	span.Finish(tracer.WithError(dbErr))
```